### PR TITLE
Feature - Interaction context subscriptions

### PR DESCRIPTION
**Pull request by atticoos**
This hits the [next stage of the roadmap](https://github.com/robinpowered/react-native-interaction-provider#roadmap) by adding subscriptions to interactivity changes.

Example:

Say we have a screen that should autodismiss after 1m of inactivity:
```js
class SomeSreen extends React.Component {
  static contextTypes = {
    interactionProvider: React.PropTypes.object.isRequired
  };

  componentDidMount() {
    this.subscription = this.context.interactionProvider.subscribeForInactivity(
      60 * 1000, 
      this.onInactive.bind(this)
    )
  }

  componentWillUnmount() {
    this.subscription.remove()
  }

  onInactive() {
    this.props.navigator.pop()
  }

  ...
}
```

This also decouples the `InteractionProvider` from the `InteractionContainer`. We may not always want both of them together, so you can now use either one. The `InteractionContainer` was for scenarios like such:

```js
<InteractionContainer timeout={60 * 1000} onInactive={() => ...}>
   <ComponentToObserveInteractionsOn />
</InteractionContainer>
```

This also introduces a convenient `withInteractionProps` component factory function, allowing components to receive `active` and `inactive` props:

```js
function InteractiveView ({inactive}) {
  const style = {backgroundColor: inactive ? 'gray' : 'red'}
  return (
    <View style={style} />
  )
}

const InteractionAwareView = withInteractionProps(InteractiveView, {duration: 60 * 1000})
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,114 +1,11 @@
...
-   */
-  render() {
-    return React.cloneElement(this.props.children, {
-      ...this.panResponder.panHandlers
-    });
-  }
-}
+import InteractionProvider from './interactionProvider';
+import InteractionContainer from './interactionContainer';
+import withInteractionProps from './withInteractionProps';
```
Comment (christophermark) [inline index.js]: These can be shortened as:
```js
export {default as InteractionContainer} from './interactionContainer';
```

File: interactionProvider.js [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+        subscribeForInactivity: this.subscribeForInactivity
+      }
+    };
+  }
+
+  /**
+   * Creates an interaction subscription that will observe periods of inactivity and when inactivity becomes resumed.
+   *
+   * @memberof InteractionContainer
+   * @param {number} duration The amount of time, in miliseconds, for a period of time to
```
Comment (christophermark) [inline interactionProvider.js]: Might be clearer as:
> The amount of time, in miliseconds, after which the interaction is considered inactive.

This comment also applies to duplicate docblocks in this PR.

File: withInteractionProps.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ */
+function withInteractionProps(WrappedComponent, {duration}) {
+  class InteractionPropertyWrapper extends React.Component {
+    static contextTypes = {
+      interactionProvider: React.PropTypes.object.isRequired
+    };
+
+    state = {
+      active: true,
+      inactive: false
```
Comment (christophermark) [inline withInteractionProps.js:27]: Are these mutually exclusive?

At first glance this seems like something that could be combined into a single prop: `active`, which is true when active and false when inactive.

**Review by christophermark — CHANGES_REQUESTED** [review]
In addition to the code changes, can you update the readme with the behavior changes?

File: index.js (hunk 1 above)
Comment (atticoos) [reply index.js]: I wanted to provide both default and named exports -- but I suppose I could just do the default one twice:

```js
export {default} from './interactionContainer'
export {default as InteractionContainer} from './interactionContainer'
...
```
Or maybe even the following might work, will have to test:
```js
export {default, default as InteractionProvider} from './interactionProvider'
```

File: withInteractionProps.js (hunk 3 above)
Comment (atticoos) [reply withInteractionProps.js:27]: Yep! But i wanted to be as ergonomic as possible to satisfy the use case where you might only care about one of these. Certainly `active` implies `inactive`, and visa-versa, but wanted to make both available for it to be as easy to use as possible

For example, in a case where we only care about `active`, we can inspect `this.props.active`. And alternatively if we only care about `inactive`, we can inspect `this.props.inactive`.

File: index.js (hunk 1 above)
Comment (christophermark) [reply index.js]: Mind replying to the thread with what did/didn't work? Really curious

File: withInteractionProps.js (hunk 3 above)
Comment (christophermark) [reply withInteractionProps.js:27]: I guess it doesn't hurt to add it, but `!active` seems pretty straightforward to me if you're detecting inactivity.

**Comment by atticoos** [comment]
Nice, the following export **did work**

```js
export {default, default as InteractionProvider} from './interactionProvider'
```

Feedback is in

**Review by christophermark — APPROVED** [review]
👍  Great work here

**Pull request merged**